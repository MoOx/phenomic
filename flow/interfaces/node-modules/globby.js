declare module "globby" {
  declare type async =
    (pattern: Array<string>, options: Object) => Promise<Array<string>>
  declare var exports: {
    (): async,
    sync: (pattern: Array<string>, options: Object) => Array<string>
  }
}
