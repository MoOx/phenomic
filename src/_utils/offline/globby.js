// @flow
import globby from "globby"

export default function(
  pattern: Array<string>,
  cwd: string,
  options?: { [key: string]: any }
): Promise {
  return globby(pattern, {
    cwd,
    nodir: true,
    ...options,
  })
}
