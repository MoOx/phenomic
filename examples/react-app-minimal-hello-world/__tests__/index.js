/* eslint-disable import/no-extraneous-dependencies */
import globby from "globby";

it("should build example correctly", () => {
  const testFolder = __dirname + "/../dist";
  const files = globby.sync("**/*", {
    cwd: testFolder,
    nodir: true
  });

  // should have html files
  const htmlFiles = files.filter(file => file.endsWith(".html"));
  expect(htmlFiles).toMatchSnapshot();

  const jsonApiFiles = files.filter(
    file => file.startsWith("phenomic") && file.endsWith(".json")
  );
  // should have matching json files
  expect(jsonApiFiles).toMatchSnapshot();

  // should have assets
  expect(
    files.filter(
      file => !htmlFiles.includes(file) && !jsonApiFiles.includes(file)
    )
  ).toMatchSnapshot();
});
