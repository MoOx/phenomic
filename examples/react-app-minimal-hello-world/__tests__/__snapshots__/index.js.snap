// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should build example correctly 1`] = `
Array [
  "index.html",
]
`;

exports[`should build example correctly 2`] = `Array []`;

exports[`should build example correctly 3`] = `
Array [
  "phenomic.main.js",
]
`;
